components/scream/src/physics/p3/CMakeLists.txt

```diff
@@ -38,6 +38,6 @@ add_library(p3 ${P3_SRCS})
 target_include_directories(p3 PUBLIC ${SCREAM_INCLUDE_DIRS} ${SCREAM_TPL_INCLUDE_DIRS} ${CIMEROOT}/src/share/include)
 set_target_properties(p3 PROPERTIES
   Fortran_MODULE_DIRECTORY ${SCREAM_F90_MODULES})
-# target_link_libraries(p3 scream_share ${SCREAM_TPL_LIBRARIES})
+target_link_libraries(p3 scream_share ${SCREAM_TPL_LIBRARIES})
```

Comment:
Just curious what's going on here?

Reply:
To be honest I am not a CMake expert, so I'm not sure.  But when I remove (or comment) this line the build in CIME breaks.

Reply:
My guess: some executable was not explicitly asking to link to scream_share. Honestly, if p3 uses scream_share (and it does), it is cleaner to have p3 "linking" scream_share. It doesn't really link (it's a library), but its dependencies are then transferred to downstream targets. This is better than having a customer of libp3.a having to figure out all of p3 deps and link them.

**tl;dr**: I think it's correct to call `target_link_libraries`.

Reply:
So I should leave this in?

Reply:
The way it is right now is correct, I think.